src/ansys/additive/core/download.py

```diff
@@ -85,3 +85,54 @@ def download_file(
                     raise ValueError(msg)
                 f.write(response.content)
     return dest
+
+
+def download_logs(
+    stub: SimulationServiceStub,
+    local_folder: str,
+    progress_handler: IProgressHandler = None,
+) -> str:
+    """Download logs from the server to the localhost.
+
+    Parameters
+    ----------
+    stub: SimulationServiceStub
+        gRPC stub for the simulation service.
+    local_folder: str
+        Folder on your localhost to write the server logs to.
+    progress_handler: ProgressLogger, None, default: None
+        Progress update handler. If ``None``, no progress will be provided.
+
+    Returns
+    -------
+    str
+        Local path of downloaded file.
+
+    """
+
+    if not os.path.isdir(local_folder):
+        os.makedirs(local_folder)
+
+    request = DownloadLogsRequest()
+    dest = os.path.join(local_folder, "AdditiveServerLogs.zip")
+
+    with open(dest, "wb") as f:
+        for response in stub.DownloadLogs(request):
+            if progress_handler:
```

Comment:
We shouldn't be duplicating code. Let's put the response handling in a separate function that can be used by both `download_file()` and `download_logs()`